docker/docker_client.go

```diff
@@ -95,8 +97,13 @@ type dockerClient struct {
 	scheme             string // Empty value also used to indicate detectProperties() has not yet succeeded.
 	challenges         []challenge
 	supportsSignatures bool
-	// Private state for setupRequestAuth
-	tokenCache map[string]bearerToken
+
+	// Private state for setupRequestAuth (key: string, value: bearerToken)
+	tokenCache sync.Map
```

Comment:
“The Map type is specialized. Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content.”

Reply:
I am not sure if you like or dislike the change. I read the docs but my impression is that the code is easier to maintain and read as those maps are only used in this file at very few places. But for sure I can change it back and add per-map locks if you prefer.

Reply:
I leave it for now as is. We can still move to use per-invariant locks or one to guard accesses to all which might be cheaper.